### PR TITLE
DDF-4329 Adding support for linestrings

### PR DESCRIPTION
**Pull request by gjvera**
#### What does this PR do?
Adds support for linestrings for geo searches 
#### Who is reviewing it? 
@GabrielFabian @mojogitoverhere @willwill96 @rymach 

#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@djblue
#### What are the relevant tickets?
[DDF-4329](https://codice.atlassian.net/browse/DDF-4329)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js [hunk 1]
```diff
@@ -334,8 +334,10 @@ function arrayFromPolygonWkt(wkt) {
   if (wkt.startsWith('POLYGON')) {
     const polygon = wkt.match(/\(\([^\(\)]+\)\)/g)
     return polygon.length === 1 ? arrayFromPartialWkt(polygon[0]) : []
+  } else if (wkt.startsWith('LINESTRING')) {
+    const lineString = ['(' + wkt.slice(10) + ')']
```
Comment (rymach) [inline ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js]: why make linestring an array if it's only ever going to have one element in it?

File: ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js [hunk 2]
```diff
@@ -334,8 +334,10 @@ function arrayFromPolygonWkt(wkt) {
   if (wkt.startsWith('POLYGON')) {
     const polygon = wkt.match(/\(\([^\(\)]+\)\)/g)
     return polygon.length === 1 ? arrayFromPartialWkt(polygon[0]) : []
+  } else if (wkt.startsWith('LINESTRING')) {
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js]: The name of this function feels a little off if it handles Linestrings as well as Polygons.

File: ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js [hunk 3]
```diff
@@ -75,7 +75,7 @@ Draw.PolygonView = Marionette.View.extend({
   coordsToLineString: function(rawCoords) {
     const coords = []
     const setArr = _.uniq(rawCoords)
-    if (setArr.length < 3) {
+    if (setArr.length < 2) {
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js]: I think polygons need at least 3 points.  Are we using the polygon drawing to do linestrings?

Comment (gjvera) [reply ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js]: ya, if we allow two points then the polygon drawing can also render the line strings

**Pull request closed without merging**